### ISSUE TITLE
FWHM/Spot size indicator

### ISSUE THREAD

**Issue by capitanevs** [oasys-elettra-kit/OASYS1-Wiser#59]
In the minutes of issue #37 we decided to put an "info" section similar to that of SRW

It shall gather infos such that:
N of Samples,
HEW Spot size
FWHM Spot size

We have to decide **where** to put it and **how.**
-Where: a)the same location as SRW (making the plot smaller)? b)In another dedicated tab?
-How: a)with a nice GUI formatting (like SRW)? b)As formatted textual output?

I am for b+a or b+b

What do you think? Some other goodideas?

![immagine](https://user-images.githubusercontent.com/13431660/131095470-42af2283-0031-47b5-a99b-c1c9484cf6d1.png)

**Comment by aljosahafner** [comment]
Let's discuss this in the meeting...

**Comment by capitanevs** [comment]
I am moving this here


Il 08/09/2021 13:52, Aljosa Hafner ha scritto:
> Yes and no, depends what is inherited and what is new for a specific optical element and I also suspect this would ruin back compatibility with ows files...
>
> On 9/8/21 1:50 PM, Michele Manfredda wrote:
>> Hi Aljosa,
>>
>> if I create a function return a dictionary, e.g.
>> {'s': 1, 'lambda (nm)': 10, 'N Samples': 10000, 'HEW (um)': 40}
>>
>> would it be easy for you to dynamially create the label + field entries in the GUI, so that you have:
>> [lambda ] ->[value]
>> [N Samples] -> [value]
>>
>> etc..
>>
>> ?

**Comment by capitanevs** [comment]
My question (a little bit out of the blue), was in the context of "spot size indicator".

I'll reframe it here?

We agreed to add a new TAB on each optical element (possible name: "Spot Info").
The initial idea (for simplicity) was to add a big text box, where to put a string coming directly from libwiser.
This string would be something like:
```

Lambda = 12nm
N Samples = 3000
HEW = 12um
FWHM = xxx um

etc

```

As an alternative, we could create a Tab where we drop "lable + text box pairs" (as in SRW).
This should be quite simple if you put extra-intelligence on the widget to handle a dictionary (or a list of lables + list of values).

Q1: apart from backcompatibility, is this as simple as it seems to me? Do you see other objections?
Q2: about backcompatibility: good point! Wouldn't the first solution (new tab + text entry) break the backcompatibility as well?
Q3: Shall we do a quick test (add new tab + text input) to test backcompatibility?

**Comment by aljosahafner** [comment]
Now I understand. This is trivial for the GUI and very easy. If it's a separate tab then I think there shouldn't be any major issues, but we need to see.

First I thought you would like to change the way GUI is constructed.

**Comment by capitanevs** [comment]
Yeah, sorry, I was a little bit cryptical :-)

**Comment by capitanevs** [comment]
I added
<OpticalElement>.ComputationData.Info

![immagine](https://user-images.githubusercontent.com/13431660/132526210-c9c89f32-2aec-4b7a-8fa6-4d204927fc50.png)

A code that takes the dictionary as input and produces a nicely formatted array of "label + text box" should nicely work with that.

**Comment by capitanevs** [comment]
(yep there is a problem in the FWHM... it is Mega-meters). The fit is wrong, I'll fix it later :-)

**Comment by aljosahafner** [comment]
Ok, excellent, I will make it tomorrow!

**Comment by capitanevs** [comment]
Yep, the code is pushed (and I hope it nicely merged with the warning suppression you did).

Perhaps... we can wait a little bit before doing that: I mean, we are just 1version away from the one that was correctly reading the ows file, so maybe we can keep the future in stanby, and inspect "the past" a little bit more :-)

**Comment by capitanevs** [comment]
Wait 3 follow up
